include/oboe/AudioStream.h

```diff
@@ -167,7 +167,7 @@ class AudioStream : public AudioStreamBase {
     * @return the resulting buffer size in frames (obtained using value()) or an error (obtained
     * using error())
     */
-    virtual ResultWithValue<int32_t> setBufferSizeInFrames(int32_t requestedFrames) {
+    virtual ResultWithValue<uint32_t> setBufferSizeInFrames(uint32_t requestedFrames) {
```

Comment:
AAudio uses signed int32_t throughout for frame counts. This was deliberate. It is Google style and prevents lots of annoying bugs.

Reply:
> I thought the reason why negative ints were used in AAudio was because they
> allowed values (positive ints) and error codes (negative ints) to be returned
> by a single type (int32_t).

No. The main reason is because there are lots of bugs that can occur when using unsigned ints and this avoids them. If you subtract two small numbers then you expect a small number. Not a huge wraparound result. Unsigned ints should only be used for non-numbers like bit masks, or if you really need the full unsigned range.

For read/write counters, for example, we want to subtract them and sometimes get a negative number if there is an underflow. Unsigned wrapping would be a mess.

> does it ever make sense for requestedFrames to be negative?

No. But it also does not make sense to have a buffer size of 1,073,741,823 frames. We don't have to use the full range of the data type.